src/solve/SimulationParameters.cpp

```diff
@@ -90,44 +198,74 @@ SimulationParameters load_simulation_params(const nlohmann::json& config) {
   sim_params.output_mean_only = sim_config.value("output_mean_only", false);
   sim_params.output_derivative = sim_config.value("output_derivative", false);
   sim_params.output_all_cycles = sim_config.value("output_all_cycles", false);
-  // DEBUG_MSG("Finished loading simulation parameters");
   return sim_params;
 }
 
-/**
- * @brief Load model from a configuration
- *
- * @param config The json configuration
- */
 void load_simulation_model(const nlohmann::json& config, Model& model) {
-  // DEBUG_MSG("Loading model");
-
   // Create list to store block connections while generating blocks
   std::vector<std::tuple<std::string, std::string>> connections;
 
   // Create vessels
-  // DEBUG_MSG("Load vessels");
   std::map<int, std::string> vessel_id_map;
-  const auto& vessels = config["vessels"];
+  if (config.contains("vessels")) {
+    create_vessels(model, connections, config["vessels"], vessel_id_map);
+  }
+
+  // Create map for boundary conditions to boundary condition type
+  std::map<std::string, std::string> bc_type_map;
+  for (const auto& bc_config : config["boundary_conditions"]) {
+    std::string bc_name = bc_config["bc_name"];
+    std::string bc_type = bc_config["bc_type"];
+    bc_type_map.insert({bc_name, bc_type});
+  }
+
+  // Create external coupling blocks
+  if (config.contains("external_solver_coupling_blocks")) {
+    create_coupling(model, connections,
```

Comment:
I would call this `create_external_coupling` to distinguish between this and coupling/connection nodes within the 0D model.